### PR TITLE
improved box visualizations with supervision

### PR DESCRIPTION
**Pull request by SkalskiP**
Hi everyone 👋🏻 

Great job! While working with the model, I noticed that the visualizations are quite simple. I decided to open a PR adding visualizations using [supervision](https://github.com/roboflow/supervision) annotators.

BEFORE:

download (1)

AFTER:

download

### CODE REVIEW

**Comment by ArgoHA** [comment]
Isn't it too much to import whole Supervision just to make visualizations nicer? Can't we just add 1 function and not import Supervision?

**Comment by SkalskiP** [comment]
I'm supervision maintainer so I'm not objective, but it's rather lightweight package. OpenCV and Pillow are two heaviest dependencies it have, and you already have them with transformers. 

I'd also want to open a PR allowing to run inference in more SDK way. Not via script but import the model. 

I plan to create D-FINE fine-tuning tutorial [here](https://github.com/roboflow/notebooks).

**Comment by ArgoHA** [comment]
I also work on some features right now (gradient accumulation during training; inference class; other small things). Maybe we can sync to not do the same thing 2 times (about inference)

**Comment by developer0hye** [comment]
@SkalskiP @ArgoHA I created [hf demo for d-fine](https://huggingface.co/spaces/developer0hye/D-FINE) with supervision!

**Comment by anyaflya** [comment]
Thanks for the implementation! The annotations are clearer this way.
